### ISSUE TITLE
NuxtLoadingIndicator doesn't ends when going back with nested route

### ISSUE THREAD

**Issue by Bananamilk452** [nuxt/nuxt#25039]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.9.0
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      devtools
- Runtime Modules:  -
- Build Modules:    -
------------------------------

also

------------------------------
- Operating System: Windows_NT
- Node Version:     v18.18.2
- Nuxt Version:     3.8.2
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  yarn@1.22.19
- Builder:          -
- User Config:      modules, pinia, components, imports, css, postcss, runtimeConfig, googleFonts, vite, typescript, ssr, veeValidate
- Runtime Modules:  @vueuse/nuxt@10.7.0, @nuxtjs/google-fonts@3.1.3, @pinia/nuxt@0.4.11, @vee-validate/nuxt@4.12.3
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-r5biil?file=pages%2Fuser.vue

### Describe the bug

1. Click 'Go To UserPage'. LoadingIndicator works well.
2. Click 'Go To User 1111'. It goes nested route /user/[id].vue. and works well.
3. But if you go to previous page (press back button), LoadingIndicator loads slowly and it doesn't ends.

**Comment by manniL** [comment]
Related: https://github.com/nuxt/nuxt/issues/24928

**Comment by agracia-foticos** [comment]
Same issue!!

**Comment by manniL** [comment]
@agracia-foticos Please avoid comments like "+1" or "same issue" without adding any helpful context. Ideally, add some context if it is missing or simply 👍🏻 the issue.

**Comment by YumaInaura** [comment]
in my case 

Same Problem in  Nuxt 3.9.0
but not nested route

# Example

when

move Page A to B 
move Page B to A

both case indicator never ending

( Same URL but different query Pages )

```vue
<NuxtLink
  :to="{ name: 'index' }"
  >A</NuxtLink
>

<NuxtLink
  :to="{ name: 'index', query: { x: 1 } }"
  >B</NuxtLink
>

```

**Comment by Bananamilk452** [comment]
@YumaInaura i think that will be fix together at #24931
